integration-tests/src/test/java/com/datastax/oss/driver/mapper/InsertIT.java

```diff
@@ -238,6 +267,12 @@ public void should_insert_entity_if_not_exists_returning_optional_asynchronously
     @Insert
     void save(Product product);
 
+    @Insert(nullSavingStrategy = NullSavingStrategy.DO_NOT_SET)
```

Comment:
In mapper 3.x, this could be configured at the mapper level ([mapper options](https://docs.datastax.com/en/developer/java-driver/3.7/manual/object_mapper/using/#mapper-options)), so all save operations could behave in a consistent way.   I think we'd also want a similar concept in the new mapper as well right?

Reply:
We can't really propagate annotations from the mapper to DAOs. A DAO is processed independently, it doesn't know which mapper it "belongs" to. In fact, more than one mapper could return a given DAO interface.

Reply:
Yes, I was also thinking about it but as Olivier mentioned - it is processed on a different level.

Reply:
I still think we should consider some concept of Session-level configuration for the mapper for things such as this.  For example, if we make `DO_NOT_SET` the default and you are on 2.1, having to make sure this is set on every operation seems like it could be overbearing and possibly error prone.

Reply:
We could propagate a default value through `MapperContext`. Then our generated DAO methods would initialize the value like this:
```java
NullSavingStrategy nullSavingStrategy =
    (context.getNullSavingStrategy() != null)
        ? context.getNullSavingStrategy()
        : /* the value from the annotation */;
```

I have a few improvements coming in [JAVA-2262](https://datastax-oss.atlassian.net/browse/JAVA-2262) that would facilitate this (I'll open the PR shortly).

Reply:
I suggest we create a follow-up ticket, since this depends on 2262 and there is already a lot going on in this PR.

Reply:
agreed :+1:

Reply:
I've created https://datastax-oss.atlassian.net/browse/JAVA-2078